cime_config/machines/config_machines.xml

```diff
@@ -2902,50 +2881,55 @@
-    <!-- Ref: https://www.olcf.ornl.gov/for-users/system-user-guides/summit/ -->
-    <!-- 1 core/socket not available for application, so 168 = 42cores*4 in smt4 mode  -->
-
-    <!-- Useful jsrun options:
-     -n	(hyphen-hyphen)nrs	Number of resource sets
-     -a	(hyphen-hyphen)tasks_per_rs	Number of tasks per resource set
-     -c	(hyphen-hyphen)cpu_per_rs	Number of CPUs per resource set. Threads per rs.
-     -g	(hyphen-hyphen)gpu_per_rs	Number of GPUs per resource set
-     -r	(hyphen-hyphen)rs_per_host	Number of resource sets per host
-     -->
-
-    <!-- Default -->
     <environment_variables>
-      <env name="COMPILER">$COMPILER</env>
-      <env name="MPILIB">$MPILIB</env>
       <env name="NETCDF_C_PATH">$ENV{OLCF_NETCDF_ROOT}</env>
       <env name="NETCDF_FORTRAN_PATH">$ENV{OLCF_NETCDF_FORTRAN_ROOT}</env>
-      <env name="NETCDF_PATH">$ENV{OLCF_NETCDF_FORTRAN_ROOT}</env>
-      <env name="NETCDFF">$ENV{OLCF_NETCDF_FORTRAN_ROOT}</env>
       <env name="ESSL_PATH">$ENV{OLCF_ESSL_ROOT}</env>
-      <env name="NETLIB_LAPACK_PATH">$ENV{OLCF_NETLIB_LAPACK_ROOT}</env>
       <env name="PGI_ACC_POOL_ALLOC">0</env>
     </environment_variables>
-    <environment_variables SMP_PRESENT="TRUE">
-      <env name="OMP_STACKSIZE">128M</env>
-    </environment_variables>
     <environment_variables mpilib="!mpi-serial">
       <env name="HDF5_PATH">$ENV{OLCF_HDF5_ROOT}</env>
       <env name="PNETCDF_PATH">$ENV{OLCF_PARALLEL_NETCDF_ROOT}</env>
     </environment_variables>
-    <!-- <environment_variables mpilib="!mpi-serial" compiler="pgigpu"> -->
-      <!-- <env name="OMPI_CXX">$SRCROOT/externals/kokkos/bin/nvcc_wrapper</env> -->
-    <!-- </environment_variables> -->
+    <environment_variables SMP_PRESENT="TRUE">
+      <env name="OMP_STACKSIZE">128M</env>
+    </environment_variables>
+    <environment_variables>
+      <env name="RS_PER_NODE">2</env>
+      <env name="CPU_PER_RS">21</env>
+      <env name="GPU_PER_RS">0</env>
+      <env name="LTC_PRT">cpu-cpu</env>
+      <env name="NUM_RS">$SHELL{echo "2*((`./xmlquery --value TOTAL_TASKS` + `./xmlquery --value TASKS_PER_NODE` - 1)/`./xmlquery --value TASKS_PER_NODE`)"|bc}</env>
+    </environment_variables>
+    <environment_variables compiler="ibmgpu">
+      <env name="RS_PER_NODE">6</env>
+      <env name="CPU_PER_RS">7</env>
+      <env name="GPU_PER_RS">1</env>
+      <env name="LTC_PRT">gpu-cpu</env>
+      <env name="NUM_RS">$SHELL{echo "6*((`./xmlquery --value TOTAL_TASKS` + `./xmlquery --value TASKS_PER_NODE` - 1)/`./xmlquery --value TASKS_PER_NODE`)"|bc}</env>
+    </environment_variables>
+    <environment_variables compiler="pgigpu">
+      <env name="RS_PER_NODE">6</env>
+      <env name="CPU_PER_RS">3</env>
+      <env name="GPU_PER_RS">1</env>
+      <env name="LTC_PRT">gpu-cpu</env>
+      <env name="NUM_RS">$SHELL{echo "6*((`./xmlquery --value TOTAL_TASKS` + `./xmlquery --value TASKS_PER_NODE` - 1)/`./xmlquery --value TASKS_PER_NODE`)"|bc}</env>
```

Comment:
Trying to understand why pgigpu uses CPU_PER_RS 3 when ibmgpu and gnugpu use 7.

Reply:
Derived from these prior settings to match `--tasks_per_rs=3`:
```
    <MAX_TASKS_PER_NODE>84</MAX_TASKS_PER_NODE>
    <MAX_TASKS_PER_NODE compiler="pgigpu">18</MAX_TASKS_PER_NODE>
    <MAX_TASKS_PER_NODE compiler="gnugpu">42</MAX_TASKS_PER_NODE>
    <MAX_MPITASKS_PER_NODE>84</MAX_MPITASKS_PER_NODE>
    <MAX_MPITASKS_PER_NODE compiler="pgigpu">18</MAX_MPITASKS_PER_NODE>
    <MAX_MPITASKS_PER_NODE compiler="gnugpu">42</MAX_MPITASKS_PER_NODE>
```
Also `ibmgpu` defaults to `smt2`, `gnugpu` and `pgigpu` default to `smt1`.

For a 1-node example with `./cime/scripts/create_test SMS.f19_g16.X`,
```
pgigpu:
jsrun -X 1 -E OMP_NUM_THREADS=1 --nrs 6 --rs_per_host 6 --tasks_per_rs 3 -d plane:3 --cpu_per_rs 3 --gpu_per_rs 1 --bind packed:smt:1 --latency_priority gpu-cpu --stdio_mode prepended

ibmgpu:
jsrun -X 1 -E OMP_NUM_THREADS=1 --nrs 6 --rs_per_host 6 --tasks_per_rs 14 -d plane:14 --cpu_per_rs 7 --gpu_per_rs 1 --bind packed:smt:2 --latency_priority gpu-cpu --stdio_mode prepended

gnugpu:
jsrun -X 1 -E OMP_NUM_THREADS=1 --nrs 6 --rs_per_host 6 --tasks_per_rs 7 -d plane:7 --cpu_per_rs 7 --gpu_per_rs 1 --bind packed:smt:1 --latency_priority gpu-cpu --stdio_mode prepended

pgi:
jsrun -X 1 -E OMP_NUM_THREADS=1 --nrs 2 --rs_per_host 2 --tasks_per_rs 42 -d plane:42 --cpu_per_rs 21 --gpu_per_rs 0 --bind packed:smt:2 --latency_priority cpu-cpu --stdio_mode prepended

ibm:
jsrun -X 1 -E OMP_NUM_THREADS=1 --nrs 2 --rs_per_host 2 --tasks_per_rs 42 -d plane:42 --cpu_per_rs 21 --gpu_per_rs 0 --bind packed:smt:2 --latency_priority cpu-cpu --stdio_mode prepended

gnu:
jsrun -X 1 -E OMP_NUM_THREADS=1 --nrs 2 --rs_per_host 2 --tasks_per_rs 42 -d plane:42 --cpu_per_rs 21 --gpu_per_rs 0 --bind packed:smt:2 --latency_priority cpu-cpu --stdio_mode prepended
```

Reply:
IIRC (which is debatable), this was an issue with the PGI pool allocator running out of memory with more tasks per node. It's a moot point now because the pool allocator is straight up broken (why we can't use PGI anymore)